::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
: Copyright (c) 2018 Mirantis Inc., Enea AB and others.
:
: All rights reserved. This program and the accompanying materials
: are made available under the terms of the Apache License, Version 2.0
: which accompanies this distribution, and is available at
: http://www.apache.org/licenses/LICENSE-2.0
::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
From: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
Date: Thu, 7 Jun 2018 21:19:07 +0200
Subject: [PATCH] linux.system.repo: Fix archive key URL

While at it, fix keystone specifics for Rocky.

Signed-off-by: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
---
 keystone/client/os_client_config/admin_identity.yml |  2 +-
 keystone/client/v3/service/heat.yml                 | 11 +++++++++++
 linux/system/repo/mcp/mirror/v1/openstack.yml       |  2 +-
 3 files changed, 13 insertions(+), 2 deletions(-)

diff --git a/keystone/client/os_client_config/admin_identity.yml b/keystone/client/os_client_config/admin_identity.yml
index d31db498..f6b6e5d6 100644
--- a/keystone/client/os_client_config/admin_identity.yml
+++ b/keystone/client/os_client_config/admin_identity.yml
@@ -9,7 +9,7 @@ parameters:
               clouds:
                 admin_identity:
                   region_name: ${_param:openstack_region}
-                  identity_api_version: '3'
+                  identity_api_version: 3
                   interface: 'internal'
                   auth:
                     username: 'admin'
diff --git a/keystone/client/v3/service/heat.yml b/keystone/client/v3/service/heat.yml
index 6c45bfe2..f1d500c7 100644
--- a/keystone/client/v3/service/heat.yml
+++ b/keystone/client/v3/service/heat.yml
@@ -6,6 +6,9 @@ parameters:
     client:
       resources:
         v3:
+          domains:
+            heat_user_domain:
+              description: 'Contains users and projects created by heat'
           roles:
             heat_stack_user:
               name: heat_stack_user
@@ -21,6 +24,14 @@ parameters:
                 service_admin:
                   name: admin
                   project_id: service
+            heat_domain_admin:
+              password: ${_param:heat_domain_admin_password}
+              email: ${_param:admin_email}
+              roles:
+                heat_stack_admin:
+                  name: admin
+                  role_domain_id: heat_user_domain
+                  domain_id: heat_user_domain
           services:
             heat:
               type: orchestration
diff --git a/linux/system/repo/mcp/mirror/v1/openstack.yml b/linux/system/repo/mcp/mirror/v1/openstack.yml
index a4a369b2..f7048f74 100644
--- a/linux/system/repo/mcp/mirror/v1/openstack.yml
+++ b/linux/system/repo/mcp/mirror/v1/openstack.yml
@@ -9,7 +9,7 @@ parameters:
           source: "deb http://mirror.mirantis.com/${_param:apt_mk_version}/openstack-${_param:openstack_version}/${_param:linux_system_codename} ${_param:linux_system_codename} main"
           architectures: ${_param:linux_system_architecture}
           clean_file: true
-          key_url: https://mirror.mirantis.com/${_param:apt_mk_version}/openstack-${_param:openstack_version}/${_param:linux_system_codename}/archive-${_param:openstack_version}.key
+          key_url: https://mirror.mirantis.com/${_param:apt_mk_version}/openstack-${_param:openstack_version}/${_param:linux_system_codename}/archive-openstack-${_param:openstack_version}.key
           pin:
           - pin: 'release o=Mirantis'
             priority: 1100
